RestartManager.js

```diff
@@ -0,0 +1,33 @@
+let NativeCodePush = require("react-native").NativeModules.CodePush;
+
+const RestartManager = (() => {
+    let _allowed = true;
+
+    function restartApp(onlyIfUpdateIsPending = false) {
+        if (_allowed) {
+            NativeCodePush.restartApp(onlyIfUpdateIsPending);
+        }
+    }
+
+    function allow() {
```

Comment:
Now that I think about it, you were right to track the state of whether a restart was requested. If an app calls this method, but an update was installed with a restart-based install mode, I'm not sure if it makes sense to trigger a restart. It seems like it makes sense for this method to toggle the flag and only call restart if an update is pending that wanted to programmatically restart the app in the first place, but couldn't because it was previously disallowed.